sklearn/neighbors/_kde.py

```diff
@@ -304,22 +303,17 @@ def sample(self, n_samples=1, random_state=None):
             cumsum_weight = np.cumsum(np.asarray(self.tree_.sample_weight))
             sum_weight = cumsum_weight[-1]
             i = np.searchsorted(cumsum_weight, u * sum_weight)
-        if self.kernel == "gaussian":
-            return np.atleast_2d(rng.normal(data[i], self.bandwidth))
 
+        size = data[i].shape
+        if self.kernel == "gaussian":
+            additional = rng.normal(0, self.bandwidth, size=size)
         elif self.kernel == "tophat":
-            # we first draw points from a d-dimensional normal distribution,
-            # then use an incomplete gamma function to map them to a uniform
-            # d-dimensional tophat distribution.
-            dim = data.shape[1]
-            X = rng.normal(size=(n_samples, dim))
-            s_sq = row_norms(X, squared=True)
-            correction = (
-                gammainc(0.5 * dim, 0.5 * s_sq) ** (1.0 / dim)
-                * self.bandwidth
-                / np.sqrt(s_sq)
-            )
-            return data[i] + X * correction[:, np.newaxis]
+            additional = rng.uniform(-self.bandwidth, self.bandwidth, size=size)
+        elif self.kernel == "exponential":
+            additional = rng.laplace(0, self.bandwidth, size=size)
+        elif self.kernel == "linear":
+            additional = rng.triangular(-self.bandwidth, 0, self.bandwidth, size=size)
```

Comment:
Same here: this works in one dimension, but I don't think it returns the correct sampling in multiple dimensions.